sound/soc/sof/intel/mtl.c

```diff
@@ -431,6 +432,11 @@ static int mtl_dsp_cl_init(struct snd_sof_dev *sdev, int stream_tag, bool imr_bo
 
 	dev_dbg(sdev->dev, "Primary core power up successful\n");
 
+	/* set enabled cores mask and increment ref count for cores in init_core_mask */
+	sdev->enabled_cores_mask |= chip->init_core_mask;
+	mask = sdev->enabled_cores_mask;
+	sdev->dsp_core_ref_count[SOF_DSP_PRIMARY_CORE]++;
```

Comment:
where is this count decreased on errors, I don't see it being handled in the err: label? What am I missing?

Reply:
thanks, that is I missed